### DISCUSSION TITLE
Can't Flash FF to Gotek Drive - ISP Gives Address/File Size Error

### DISCUSSION THREAD

**Discussion by Raven2k360** [keirf/flashfloppy#674] [Q&A] [answered]
Hi All,
   Hopefully someone will be able to help me sort this issue out.  I purchased a couple Gotek drives to try using on some equipment at work (Inficon IC5 Deposition Controller) as a cheaper alternative to the 'Richpeace' units.  The Gotek drives won't work out of the box, so I stumbled across the FF wiki and reflashing them to possibly work.  I've downloaded FF 3.34 and the Artery ISP (that link is broken, BTW) and I got as far as setting up to flash the FF hex file to the drive.  When I click 'Next' to start the flashing, I get a Warning pop-up stating, **"The file download address or file size out of flash range!"**

  The PCB revision is SFRKC30.AT2, and the unit as a whole is the one without the LCD display or buttons.  It also has the AT32F415 chip (ID of the 'Target' is AT32F415KBU7-4_128K), and that is the hex file I'm attempting to use.  The programmer does auto-detect the Gotek and I was able to successfully disable the protection during that step (first time did fail for some reason, but the second attempt worked).

   Other information that may be pertinent:
PID (h): 700301C6
BID (h): 4602
Protocol Version: 3.2
Flash Mapping: Sector 0 @ Start Address 0x08000000, Sector 127 @ End Address 0x0801FFFF, Sector Sizes 0x400(1K)

   The ISP is showing the following information on the hex file before flashing:
File Size: 111320
Address Range(Ox): 08000000-08007083, 0800C00-08020253

Any help would be greatly appreciated in getting this resolved.  After that, I may need some help trying to get the drive to read still in the equipment (another post for another time).

**Comment by keirf** [comment, marked as answer]
You're probably using the wrong hex file.

**Reply by Raven2k360** [reply]
Well, you are correct sir...  I suspected it was something with the hex file, and after looking further, I found the A32 file I was using was for the F35 chip, not the F15 - the F15 is part of the STM file.  I apparently read too fast and the brain didn't catch up...D'OH!  Thank you for responding, I appreciate it.  Flashing was successful, so now off to test some more!

**Reply by keirf** [reply]
No problem. I think I will change the naming to `flashfloppy_at415_st105_xxx.hex` next release. More obvious there are two Artery options, and more people will be programming AT32F415 than STM32F105 these days.

**Comment by keirf** [comment]
Thanks for mentioning the broken link. Now fixed. I also updated instructions because Microsoft Defender warns about the unsigned exe, and because this latest version defaults to Chinese.
